pkg/R/sparkR.R

```diff
@@ -149,11 +156,14 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+        length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    assign(".monitorConn", socketConnection(port = monitorPort), envir = .sparkREnv)
```

Comment:
In addition to this, a couple of things can be removed now:
- Line 217 (`reg.finalizer`) can be removed as we don't need `sparkR.stop` to be run when the R process exits
- Line 28-37 can be refactored as follows: We check if a valid connection exists and if it doesn't we can just exit saying no connection to backend found (no need to reconnect). Also then we don't the `if` in Line 40 and the whole block can just go inside `if (connExists)`

@sun-rui does this sound good ?